dropwizard-jooq/src/main/java/com/bendb/dropwizard/jooq/jersey/JooqBinder.java

```diff
@@ -1,35 +1,58 @@
 package com.bendb.dropwizard.jooq.jersey;
 
+import org.glassfish.hk2.api.InjectionResolver;
+import org.glassfish.hk2.api.TypeLiteral;
 import org.glassfish.hk2.utilities.binding.AbstractBinder;
 import org.glassfish.jersey.process.internal.RequestScoped;
+import org.glassfish.jersey.server.spi.internal.ValueFactoryProvider;
 import org.jooq.Configuration;
 import org.jooq.ConnectionProvider;
 import org.jooq.DSLContext;
 
+import javax.inject.Singleton;
+import java.util.Map;
+
 /**
  * Create bindings for {@link DSLContext} (via
  * {@link com.bendb.dropwizard.jooq.jersey.DSLContextFactory}),
  * {@link org.jooq.Configuration}, and {@link org.jooq.ConnectionProvider}.
+ *
+ * Also creates bindings for {@link com.bendb.dropwizard.jooq.jersey.DataSourceParam} (via
+ * {@link com.bendb.dropwizard.jooq.jersey.DataSourceParamValueFactoryProvider} and
+ * {@link com.bendb.dropwizard.jooq.jersey.DataSourceParamInjectionResolver}.
  */
 public class JooqBinder extends AbstractBinder {
 
-    private final Configuration configuration;
+    private final Map<String,Configuration> configurationMap;
```

Comment:
I've implemented a LinkedHashMap to take care of ordering for now. This cleans up the excess code a bit - I will experiment with the enum idea to see if this might be a better solution.